include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+#ifndef _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+#define _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+
+#include <type_traits>
+#include "nbl/asset/utils/IGLSLCompiler.h" // asset::IGLSLCompiler::E_SPIRV_VERSION
+
+
+namespace nbl::video
+{
+
+struct SPhysicalDeviceLimits
+{
+    /* Vulkan 1.0 Core  */
+    uint32_t maxImageDimension1D;
+    uint32_t maxImageDimension2D;
+    uint32_t maxImageDimension3D;
+    uint32_t maxImageDimensionCube;
+    uint32_t maxImageArrayLayers;
+    uint32_t maxBufferViewSizeTexels;
+    uint32_t maxUBOSize;
+    uint32_t maxSSBOSize;
+    //uint32_t              maxPushConstantsSize;
+    //uint32_t              maxMemoryAllocationCount;
+    //uint32_t              maxSamplerAllocationCount;
+    //VkDeviceSize          bufferImageGranularity;
+    //VkDeviceSize          sparseAddressSpaceSize;
+    //uint32_t              maxBoundDescriptorSets;
+    //uint32_t              maxPerStageDescriptorSamplers;
+    //uint32_t              maxPerStageDescriptorUniformBuffers;
+    uint32_t maxPerStageDescriptorSSBOs;
+    //uint32_t              maxPerStageDescriptorSampledImages;
+    //uint32_t              maxPerStageDescriptorStorageImages;
+    //uint32_t              maxPerStageDescriptorInputAttachments;
+    //uint32_t              maxPerStageResources;
+    //uint32_t              maxDescriptorSetSamplers;
```

Comment:
I think we want to expose these too.

Reply:
yes ofc!

Reply:
you also want to add the validation during PipelineLayoutCreation I guess (the limits are across all descriptor sets) 

> Only descriptors in descriptor set layouts created without the VK_DESCRIPTOR_SET_LAYOUT_CREATE_UPDATE_AFTER_BIND_POOL_BIT bit set count against this limit.

### Per Stage
> A descriptor is accessible to a shader stage when the stageFlags member of the VkDescriptorSetLayoutBinding structure has the bit for that shader stage set.

Reply:
I want to set these in GL:
```
    //uint32_t              maxPerStageDescriptorSamplers; // DON'T EXPOSE: we only support combined image samples
    //uint32_t              maxPerStageDescriptorUniformBuffers;
    uint32_t maxPerStageDescriptorSSBOs; // already exposed in GL and basically a minimum of bunch of MAX_XXX_SHADER_STORAGE_BLOCK
    //uint32_t              maxPerStageDescriptorSampledImages;
    //uint32_t              maxPerStageDescriptorStorageImages;
    //uint32_t              maxPerStageDescriptorInputAttachments;
    //uint32_t              maxPerStageResources;
```
Please VERIFY @devshgraphicsprogramming :
I guess I can simply set those the non per stage ones in GL? because we already implement these:
```c++
m_properties.limits.maxDescriptorSetUBOs = m_glfeatures.maxUBOBindings;
m_properties.limits.maxDescriptorSetDynamicOffsetUBOs = SOpenGLState::MaxDynamicOffsetUBOs;
m_properties.limits.maxDescriptorSetSSBOs = m_glfeatures.maxSSBOBindings;
m_properties.limits.maxDescriptorSetDynamicOffsetSSBOs = SOpenGLState::MaxDynamicOffsetSSBOs;
m_properties.limits.maxDescriptorSetImages = m_glfeatures.maxTextureBindings;
m_properties.limits.maxDescriptorSetStorageImages = m_glfeatures.maxImageBindings;
```

Reply:
@devshgraphicsprogramming